### PR TITLE
fix: added valid url check in callback url redirection

### PR DESCRIPTION
**Pull request by devcer**
### Context

- fixes https://github.com/interledger/testnet/issues/1136

### Changes

Added an invalid URL check before the callback URL redirection

### CODE REVIEW

**Comment by devcer** [comment]
@raducristianpopa @Tymmmy  The `new URL(...).searchParams.set(...)` already encode the URL, Hence encodeURIComponent is not needed in middleware. Otherwise, it is double encoding the already encoded URL, causing other URL issues.  To verify this, you could open `https://rafiki.money/transactions` in incognito and you will see the encoded callback param in the URL.

File: packages/wallet/frontend/src/pages/auth/login.tsx [hunk 1]
```diff
@@ -32,7 +32,14 @@ const LoginPage: NextPageWithLayout = () => {
             const response = await userService.login(data)
 
             if (response.success) {
-              router.push(callBackUrl)
+              const isInvalidCallbackUrl =
+                !callBackUrl.startsWith('/') &&
+                !callBackUrl.startsWith(window.location.origin)
+              if (isInvalidCallbackUrl) {
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/auth/login.tsx:38]: suggestion -> isIncorrectCallbackUrl ? router.push('/') : router.push(callbackUrl)

File: packages/wallet/frontend/src/pages/auth/login.tsx [hunk 2]
```diff
@@ -32,7 +32,14 @@ const LoginPage: NextPageWithLayout = () => {
             const response = await userService.login(data)
 
             if (response.success) {
-              router.push(callBackUrl)
+              const isInvalidCallbackUrl =
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/auth/login.tsx:35]:
```suggestion
              const isIncorrectCallbackUrl =
```

Comment (Tymmmy) [reply packages/wallet/frontend/src/pages/auth/login.tsx:35]: Trying to use politically correct wording :D

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**